Fix snapshot regular expression

kevin-canadian opened the pull request:
The snapshot regular expression should only match an explicit period.

khmarbaise commented:
First thanks for your pull request but it is really needed that you open a [JIRA ticket ](issues.apache.org/jira/browse/MNG) and furthermore it would be great if you could offer some unit tests which show the old behaviour and the new one which fix it by the appropriate fix.  Apart from that the commit message should contain a reference to the appropriate JIRA ticket cause it's can't followed later in the history and can't be seen in the release notes...

kevin-canadian commented:
I actually just wanted to let you know about the bug, and since the fix was so simple I made a PR rather than creating an issue.

It will be a while before I have time look into the project's contribution guidelines or figure out how to add and run appropriate tests.

If someone would like to incorporate this fix in the meantime that would be great, otherwise I'll try to get around to it eventually!

khmarbaise commented:
If you will take some time it's not a problem...the PR will be kept here for a long time until you will close it yourself ..so you can be patient and take the time you need...no hurry needed...Furthermore it would be great if you have identified an error of any kind of or just realised by reading the code the issue? The circumstances are important...

kevin-canadian commented:
Thanks for the quick reply.

This PR was prompted by a bug we discovered in our production build environment.  When we updated our release artifact version number format to be <major>.<minor>.<patch>-YYYYMMDD-HHmmss-<build-number>, release artifacts were erroneously being sent to snapshot repositories, resulting in strange issues in our Archiva repository.

The reason appears to be that the regex used to identify timestamp-tagged snapshot version numbers erroneously uses a wildcard "." instead of a literal "." to match the separator between date and time.  This results in our version format, which uses a "-" as the separator, being incorrectly identified as a timestamp-tagged snapshot version number instead of a release version number.

For now we just modified our release artifact version number format to something else.  When I find the time I'll create a JIRA issue and a proper patch for this issue.

khmarbaise commented:
It would be great if you could do that, cause it should be honoured to those who found the bug(s)...If I can support you please ask how?

kevin-canadian commented:
If I run into any problems I'll mention them here, thanks.

khmarbaise commented:
If you like to get your pull request being integrated it is required to open an [issue on jira](https://issues.apache.org/jira/browse/MNG) and change your commit message accordingly. Something like:
```
[MNG-XXXX] ..
 o description if needed
```

jira-importer commented:
Resolve #8211
